src/main/java/org/assertj/core/api/AbstractBigDecimalAssert.java

```diff
@@ -263,6 +265,26 @@ public SELF isNotEqualByComparingTo(String notExpected) {
     return isNotEqualByComparingTo(new BigDecimal(notExpected));
   }
 
+  /**
+   * Verifies the scale of the BigDecimal passed is equal to the given one.
+   * <p>
+   * Example:
+   * <pre><code class='java'> // assertions will pass
+   * assertThat(new BigDecimal(&quot;8.00&quot;)).hasScaleOf(2);
+   * assertThat(new BigDecimal(&quot;8.00&quot;).setScale(4)).hasScaleOf(4);
+   *
+   * // assertion will fail
+   * assertThat(new BigDecimal(&quot;8.00&quot;)).hasScaleOf(3);
+   * assertThat(new BigDecimal(&quot;8.00&quot;).setScale(4)).hasScaleOf(2);</code></pre>
+   * @param expected the expected scale value..
+   * @return {@code this} assertion object.
+   */
+  public SELF hasScaleOf(int expected) {
+    isNotNull();
+    bigDecimals.assertEqual(info, actual.scale(), expected);
```

Comment:
introduce `BigDecimals.asserthasScale` method so that we can throw an assertion error that mentions scale instead of a basic equal error.
This means creating a `ShouldHaveScale` error and throwing it (see https://github.com/assertj/assertj-core/blob/main/src/main/java/org/assertj/core/error/ShouldHaveMessage.java for an example)
You would have to test `ShouldHaveScale` too ;-)

Reply:
> > On a side note, would it be cool to also add assertion methods like isBetween isStrictlyBetween for scales on BigDecimals too?
> 
> I would prefer to have a `scale` navigation method to be able to chain any number assertions to avoid duplcating them, ex:
> 
> ```java
> assertThat(bigDecimal).scale().isBetween(1, 5);
> ```
> 
> We won't have `scale` mentioned in the error message though but it's ok in this case.


Makes sense. Maybe I'll put up a separate PR for that.